File: EndGate/tests/EndGate.Core.Tests/Assets/Vectors/Vector2dHelperFacts.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using EndGate.Core.Assets;
using Xunit;

namespace EndGate.Core.Tests.Assets.Vectors
{
    public class Vector2dHelperFacts
    {
        [Fact]
        public void GetMinMaxProjectionsWorks()
        {
            var axis = new Vector2d(8, 0);
            var vertices = new Vector2d[] {
                new Vector2d(2, 0),  // TL
                new Vector2d(5, 3),  // TR
                new Vector2d(0, -2), // BL
                new Vector2d(3, 5)   // BR
            };

            double expectedMax = vertices[1].ProjectOnto(axis).Dot(axis);
            double expectedMin = vertices[2].ProjectOnto(axis).Dot(axis);

            Tuple<double, double> maxmin = Vector2dHelpers.GetMinMaxProjections(axis, vertices);

            Assert.Equal(maxmin.Item1, expectedMin);
            Assert.Equal(maxmin.Item2, expectedMax);
        }
    }
}

File: EndGate/EndGate.Core.Client.JS.Tests/Tests/UnitTests/Assets/Vectors/Vector2dHelperFacts.js
﻿(function (window, lib) {

    QUnit.module("Vector2d Helper Facts");

    QUnit.test("Get Min Max projections works", function () {
        var axis = new lib.Vector2d(8, 0);
        var vertices = [
            new lib.Vector2d(2, 0),  // TL
            new lib.Vector2d(5, 3),  // TR
            new lib.Vector2d(0, -2), // BL
            new lib.Vector2d(3, 5)   // BR
        ];

        var expectedMax = vertices[1].ProjectOnto(axis).Dot(axis);
        var expectedMin = vertices[2].ProjectOnto(axis).Dot(axis);

        var maxmin = lib.Vector2dHelpers.GetMinMaxProjections(axis, vertices);

        QUnit.ok(maxmin.Min == expectedMin);
        QUnit.ok(maxmin.Max == expectedMax);
    });

})(window, EndGate.Core.Assets);